### ISSUE TITLE
How to compress or decompress data?

### ISSUE THREAD

**Issue by patlecat** [microsoft/cpprestsdk#1419]
When I send the following header line to force compression, then I can't find a function to decompress the response.

```
crequest.headers().add(header_names::accept_encoding, L"br, deflate, gzip");
```

**Comment by patlecat** [comment]
No one knows?

**Comment by fxfire** [comment]
thats just the header, it has nothing to do with the actual process of (de-)compression.

have a look at the response and request classes, there is the stuff you need afaik

**Comment by patlecat** [comment]
Yeah and it's in the header that you can request compression. So where exactly and how can I compress/decompress the content/response?

**Comment by fxfire** [comment]
https://github.com/microsoft/cpprestsdk/blob/master/Release/tests/functional/http/client/compression_tests.cpp

**Comment by patlecat** [comment]
Sorry man but that's not helpful. I can browse the codefiles too. I need answers from an experienced user on how this is supposed to work in this lib.

**Comment by fxfire** [comment]
Maybe this gives you a start ?
https://github.com/microsoft/cpprestsdk/issues/1405

**Comment by patlecat** [comment]
Ahh yes that might help. I'll check it out. Thanks.

**Comment by MakotoE** [comment]
In case someone comes across this on Google: check out https://github.com/microsoft/cpprestsdk/issues/226#issuecomment-586301793. All you need to do is:

```
http_client_config config; 
config.set_request_compressed_response(true); 
```

And `Accept-Encoding` in the request becomes `gzip;q=0.500, deflate;q=0.500, br;q=0.500`.

**Comment by patlecat** [comment]
> 
> 
> In case someone comes across this on Google: check out [#226 (comment)](https://github.com/microsoft/cpprestsdk/issues/226#issuecomment-586301793). All you need to do is:
> 
> ```
> http_client_config config;
> config.set_request_compressed_response(true);
> ```
> 
> And `Accept-Encoding` in the request becomes `gzip;q=0.500, deflate;q=0.500, br;q=0.500`. It's a shame the method is not visible in the docs.

Ok I'll try that, thanks. Yeah documentation is not helpful.
